services/cases-api/lambdas/updateCase.js

```diff
@@ -25,32 +27,44 @@ export async function main(event) {
     return response.failure(new BadRequestError('Missing required path parameter "id"'));
   }
 
-  const { error, value: validatedEventBody } = updateCaseValidationSchema.validate(
-    requestJsonBody,
-    {
-      abortEarly: false,
-    }
-  );
+  const { error, value: validatedJsonBody } = updateCaseValidationSchema.validate(requestJsonBody, {
+    abortEarly: false,
+  });
   if (error) {
     return response.failure(new BadRequestError(error.message.replace(/"/g, "'")));
   }
 
-  const { currentFormId, currentPosition, answers, signature } = validatedEventBody;
+  const [getUserCaseError, userCase] = await to(getUserCase(personalNumber, id));
+  if (getUserCaseError) {
+    return response.failure(new InternalServerError(getUserCaseError));
+  }
 
-  const [getCaseError] = await to(getCaseWhereUserIsApplicant(id, personalNumber));
-  if (getCaseError) {
-    return response.failure(
-      new ResourceNotFoundError('The user does not have any case with the provided case id')
-    );
+  if (!userCase) {
+    return response.failure(new ResourceNotFoundError('Case not found'));
   }
 
-  const newCaseStatus = getNewCaseStatus(answers, signature);
+  const { currentFormId, currentPosition, answers, signature } = validatedJsonBody;
 
-  // Create update expression
-  const UpdateExpression = ['SET updatedAt = :newUpdatedAt'];
+  const UpdateExpression = ['updatedAt = :newUpdatedAt'];
   const ExpressionAttributeNames = {};
   const ExpressionAttributeValues = { ':newUpdatedAt': Date.now() };
 
+  const updatedUserCasePeople = userCase.persons?.map(person => {
+    const newPerson = { ...person };
+    if (newPerson.personalNumber === personalNumber) {
+      newPerson.hasSigned = signature?.success || false;
```

Comment:
In case `signature` is not provided, this would reset it to `false`. Is this intended?

Reply:
Good point, do not know if this is behaviour is wanted. My thought is that if a user updates the answers on a certain case, we want to reset the signature since the answers is updated and therefore the signature is for a previous collection of answers.

Reply:
In addition to this we probably also want to reset all signatures, if there is a case where several users have signed a case and an update is made that makes the previous case state outdated.

Reply:
Not sure what to do with this info? Help me understand what to do. :)

Reply:
> Not sure what to do with this info? Help me understand what to do. :)

Something like this maybe?

```javascript
const updatedUserCasePeople = userCase.persons?.map(person => {
  const newPerson = { ...person };

  if (newPerson.personalNumber === personalNumber && signature) {
    newPerson.hasSigned = signature.success;
  } else if (answers) {
    newPerson.hasSigned = false;
  }

  return newPerson;
});
```

use-case 1: no answers, no signature (everything stays the same)
use-case 2: answers, no signature (all signatures are reset)
use-case 3: answers, signature (signature is updated for personalNumber provided, all others are reset)
use-case 4: no answers, signature (signature is updated for provided personalNumber only)

Reply:
Done!

Reply:
Fixed